File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int max = 0;
        
        for (int num: piles) {
            max = Math.max(max, num);
        }

        int k = max;
        int left = 1, right = k;

        while (left <= right) {
            int mid = (left + right) / 2;

            int hours = 0;
            for (int num: piles) {
                hours += Math.ceil((double) num / mid);
            }

            if (hours <= h) {
                k = Math.min(k, mid);
                right = mid - 1;
            }
            else {
                left = mid + 1;
            }
        }

        return k;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        setNums = set(nums)
        longest = 0

        for num in setNums:
            if (num - 1) not in setNums:
                length = 1
                
                while (num + length) in setNums:
                    length += 1

                longest = max(longest, length)

        return longest
        